erpnext/public/js/projects/timer.js

```diff
@@ -0,0 +1,135 @@
+frappe.provide("erpnext.timesheet");
+
+erpnext.timesheet.timer = function(frm, row, timestamp=0) {
+	let dialog = new frappe.ui.Dialog({
+		title: __("Timer"),
+		fields:
+		[
+			{"fieldtype": "Link", "label": __("Activity Type"), "fieldname": "activity_type",
+				"reqd": 1, "options": "Activity Type"},
+			{"fieldtype": "Link", "label": __("Project"), "fieldname": "project", "options": "Project"},
+			{"fieldtype": "Float", "label": __("Expected Hrs"), "fieldname": "expected_hours"},
+			{"fieldtype": "Section Break"},
+			{"fieldtype": "HTML", "fieldname": "timer_html"}
+		]
+	});
+
+	if (row) {
+		dialog.set_values({
+			'activity_type': row.activity_type,
+			'project': row.project,
+			'expected_hours': row.expected_hours
+		});
+	}
+	dialog.get_field("timer_html").$wrapper.append(frappe.render_template("timesheet"));
+	control_timer(frm, dialog, row, timestamp);
+	dialog.show();
+};
+
+var control_timer = function(frm, dialog, row, timestamp=0) {
```

Comment:
Will cause this to be left hanging as a global. Let's namespace it with `erpnext.timesheet` as well :)